### ISSUE TITLE
When using Filter, Balance/Changes screen should not show accounts which have zero value because of the filter

### ISSUE THREAD

**Issue by prafullat** [beancount/fava#1116]
See this URL for example:
https://fava.pythonanywhere.com/huge-example-file/account/Expenses/balances/?filter=payee%3A%22Uncle+Boons%22

Uncle Boons filter is showing values for only Food and Restaurant, If we skip showing other accounts, it would be better.

**Comment by yagebu** [comment]
There's an option for that: `show-accounts-with-zero-balance`

**Comment by prafullat** [comment]
I used both of the following options, but it still showed empty accounts on balance/changes screen ( above URL)

```
1970-01-01 custom "fava-option" "show-accounts-with-zero-balance" "false"
1970-01-01 custom "fava-option" "show-accounts-with-zero-transactions" "false"
```

**Comment by yagebu** [comment]
You're quite right. These options are currently not respected in the balance and change account trees. A PR fixing that would be welcome.

**Comment by prafullat** [comment]
Ok, I would like to help with it. Can you please assign it to me?

**Comment by prafullat** [comment]
Hi, It looks like, it started working now. Not sure what has changed, but I do not see accounts which are zero or no transactions.  Can you please confirm if any recent changes fix it?

**Comment by prafullat** [comment]
Added a fix in this PR : https://github.com/beancount/fava/pull/1118
Please take a look

**Comment by xiaoyuesansui** [comment]
> I used both of the following options, but it still showed empty accounts on balance/changes screen ( above URL)
> 
> ```
> 1970-01-01 custom "fava-option" "show-accounts-with-zero-balance" "false"
> 1970-01-01 custom "fava-option" "show-accounts-with-zero-transactions" "false"
> ```

hello mate, tried above lines on June 2023 ,Fava version 1.23.x 
it works well on my end 

maybe someone fix the issues ?

**Comment by yagebu** [comment]
This will be fixed by #1664
